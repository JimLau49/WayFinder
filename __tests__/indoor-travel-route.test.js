import React from "react";
import IndoorTravelRoute from "../src/components/travel-route/indoor-travel-route.component";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import { Alert } from "react-native";
import * as testData from "./__testData__/indoor-travel-route";

describe("Indoor Travel Route component", () => {
  it("should match indoor travel route snapshot", () => {
    const tree = renderer
      .create(
        <IndoorTravelRoute
          start={testData.mockPOIs[0]}
          end={testData.mockPOIs[1]}
          chosenFloorLevel={8}
          setTravelState={null}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("it should alert the user if theirs no indoor floor plans", () => {
    const mockSetTravelState = jest.fn();
    const wrapper = shallow(
      <IndoorTravelRoute
        start={testData.mockBuilding[0]}
        end={testData.mockBuilding[1]}
        chosenFloorLevel={8}
        setTravelState={mockSetTravelState}
      />
    );

    expect(mockSetTravelState).toHaveBeenCalledTimes(1);
  });
});
