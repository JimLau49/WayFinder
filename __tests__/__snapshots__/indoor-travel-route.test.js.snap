// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indoor Travel Route component should match indoor travel route snapshot 1`] = `
Array [
  <AIRMapPolyline
    coordinates={
      Array [
        Object {
          "latitude": 45.497253,
          "longitude": -73.579251,
        },
        Object {
          "latitude": 45.49730597420072,
          "longitude": -73.57923445639382,
        },
      ]
    }
    fillColor="#AA2B45"
    lineCap="round"
    lineDashPattern={
      Array [
        0.5,
        0.5,
      ]
    }
    lineJoin="round"
    strokeColor="#AA2B45"
    strokeWidth={5}
    style={
      Object {
        "zIndex": 2,
      }
    }
    testID="polylinePath0"
  />,
  <AIRMapPolyline
    coordinates={
      Array [
        Object {
          "latitude": 45.49730597420072,
          "longitude": -73.57923445639382,
        },
        Object {
          "latitude": 45.49735509062119,
          "longitude": -73.57918419446206,
        },
      ]
    }
    fillColor="#AA2B45"
    lineCap="round"
    lineDashPattern={
      Array [
        0.5,
        0.5,
      ]
    }
    lineJoin="round"
    strokeColor="#AA2B45"
    strokeWidth={5}
    style={
      Object {
        "zIndex": 2,
      }
    }
    testID="polylinePath1"
  />,
  <AIRMapPolyline
    coordinates={
      Array [
        Object {
          "latitude": 45.49735509062119,
          "longitude": -73.57918419446206,
        },
        Object {
          "latitude": 45.49743264359791,
          "longitude": -73.57911244537091,
        },
      ]
    }
    fillColor="#AA2B45"
    lineCap="round"
    lineDashPattern={
      Array [
        0.5,
        0.5,
      ]
    }
    lineJoin="round"
    strokeColor="#AA2B45"
    strokeWidth={5}
    style={
      Object {
        "zIndex": 2,
      }
    }
    testID="polylinePath2"
  />,
  <AIRMapPolyline
    coordinates={
      Array [
        Object {
          "latitude": 45.49743264359791,
          "longitude": -73.57911244537091,
        },
        Object {
          "latitude": 45.497482,
          "longitude": -73.579034,
        },
      ]
    }
    fillColor="#AA2B45"
    lineCap="round"
    lineDashPattern={
      Array [
        0.5,
        0.5,
      ]
    }
    lineJoin="round"
    strokeColor="#AA2B45"
    strokeWidth={5}
    style={
      Object {
        "zIndex": 2,
      }
    }
    testID="polylinePath3"
  />,
]
`;
